### DISCUSSION TITLE
Speed up sample rate ADC (ADS1115) with a EasyMCP library

### DISCUSSION THREAD

**Discussion by Jur-I2C** [electronicayciencia/EasyMCP2221#14] [General]
Hi,
Last day I connected an ADC converter (ADS1115)  to an MCP2221 breakout board.  I want to read out one channel of the ADC with a 16 bit resolution.   I found some nice ADS 1115 libraries and I could write a python program which grabs the data of channel A0.  So far so good.  However my datarate is only 33Hz.  (I get 33 samples per second)

I analyzed the SDA signal (yellow line) and SDL signal (purple line) with a scope when it sends data to the laptop : 

![SDS00018](https://github.com/user-attachments/assets/04413718-05d4-4e6e-a516-1c78b84c0778)

To my surprise there is a lot of space between the information packets.  (6 DIV = 6 miliseconds).   Is there a way to get those bits closer together ?  This will increase the datarate.   Or is this caused by the limited speed of the USB port  (max 500 Hz according the info on github) ?  I do know that the data is sent in chunks of 60 bytes in the EasyMCP2221 library while in my case I need a max. of 4 bytes.  Is it an option to set the package size for 64 to 8 bytes ?

Does anyone a good idea how to increase the data rate ?

**Comment by electronicayciencia** [comment]
It is a limitation of all USB to I2C chips. See [I2C speed limit](https://easymcp2221.readthedocs.io/en/latest/limitations_bugs.html#i2c-speed-limit) and [Internal details](https://easymcp2221.readthedocs.io/en/latest/internals.html).

Using continuous reading you could speed it up until 84 sps [ADS_continuous.py](https://github.com/chandrawi/ADS1x15-ADC/blob/main/examples/ADS_continuous.py).
